### ISSUE TITLE
Do region resolution in cloud-mirror

### ISSUE THREAD

**Issue by jonasfj** [taskcluster/cloud-mirror#6]
Move `EC2RegionResolver` from taskcluster-queue in here, see:
https://github.com/taskcluster/taskcluster-queue/blob/2ab0317576d9b7c6ca9b011a76ff36aaf7eaff3e/queue/ec2regionresolver.js

**Comment by jhford** [comment]
there should be a new route that uses `/v1/redirect/base64(inputurl)` which picks whichever EC2 region the request is coming from and uses that for the redirect instead of a `:region` and `:service` parameter
For non-ec2 results, it should redirect to just the input url

**Comment by jonasfj** [comment]
For non-ec2 it could also redirect to cloudfront... Unless of course it makes sense for us to use cloudfront for copying... I'm not sure, since we copy out, rather than in... So we would be paying for bandwidth that is otherwise free.

Note: the cloudfront thing is really just a URL rewrite nothing major...

**Comment by jhford** [comment]
Yep!  We could totally do that.  For now, let's do the safe thing and maybe we could share the URL rewriting logic in the api with #10.

**Comment by jonasfj** [comment]
Yeah, sorry... I was really talking about doing #10.
What i mean is we can't move region resolution out of the queue until both are done...

But #10 is a very tiny thing to do. It probably has to tie in with IP lookups anyways...

**Comment by jhford** [comment]
You're right.  It would be great if we could have something that takes an IP address and returns `{service: 'cloudfront'|'s3', region: 'us-west-1}` or something that could be used to decide how to rewrite the URL.
